src/cast_core/test/MultiCollectionInsert_test.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/json.hpp>
+
+#include <boost/exception/diagnostic_information.hpp>
+
+#include <yaml-cpp/yaml.h>
+
+#include <testlib/ActorHelper.hpp>
+#include <testlib/MongoTestFixture.hpp>
+#include <testlib/helpers.hpp>
+
+#include <gennylib/context.hpp>
+
+namespace genny {
+namespace {
+using namespace genny::testing;
+namespace bson_stream = bsoncxx::builder::stream;
+
+TEST_CASE_METHOD(
+    MongoTestFixture, "MultiCollectionInsert", "[standalone][MultiCollectionInsert]") {
+
+    dropAllDatabases();
+    auto db = client.database("mydb");
+
+    NodeSource nodes = NodeSource(R"(
+        SchemaVersion: 2018-07-01
+        Actors:
+        - Name: MultiCollectionInsert
+          Type: MultiCollectionInsert
+          Database: mydb
+          Phases:
+          - Repeat: 100
+            Collection: mycoll
+            Document: {foo: {^RandomInt: {min: 0, max: 100}}}
+    )",
+                                  __FILE__);
+
+
+    SECTION("Inserts documents into the database.") {
+        try {
+            genny::ActorHelper ah(nodes.root(), 1, MongoTestFixture::connectionUri().to_string());
+            ah.run([](const genny::WorkloadContext& wc) { wc.actors()[0]->run(); });
+
+            auto builder = bson_stream::document{};
+            builder << "foo" << bson_stream::open_document << "$gte"
+                    << "0" << bson_stream::close_document << bson_stream::finalize;
+
+            auto count = db.collection("mycoll").count_documents(builder.view());
+            // TODO: fixme
+            REQUIRE(count == 101);
```

Comment:
I think this will fail? If not it..should..I think. At any rate I think for this to be correct you would need to set `CollectionCount` to some reasonable number and then do the `count_documents` over each of the possible collections.

Reply:
Is the test not being run as part of the commit? I'll update to the test now anyway.